src/frontends/tensorflow/src/op/reverse.cpp

```diff
@@ -3,47 +3,140 @@
 //
 
 #include "op_table.hpp"
-#include "openvino/opsets/opset8.hpp"
+#include "openvino/opsets/opset9.hpp"
 
 using namespace std;
-using namespace ov::opset8;
+using namespace ov;
+using namespace ov::opset9;
 
 namespace ov {
 namespace frontend {
 namespace tensorflow {
 namespace op {
+shared_ptr<Node> compute_sequence_lengths(const Output<Node>& input_shape, int64_t batch_axis, int64_t seq_axis) {
+    auto batch_axis_const = make_shared<Constant>(element::i64, Shape{1}, batch_axis);
+    auto seq_axis_const = make_shared<Constant>(element::i64, Shape{1}, seq_axis);
+    auto gather_axis = make_shared<Constant>(element::i32, Shape{}, 0);
+    auto batch_dim = make_shared<Gather>(input_shape, batch_axis_const, gather_axis);
+    auto seq_dim = make_shared<Gather>(input_shape, seq_axis_const, gather_axis);
+    auto seq_lengths = make_shared<Broadcast>(seq_dim, batch_dim);
+
+    return seq_lengths;
+}
+
+OutputVector translate_reverse_base_op(const NodeContext& node,
+                                       const Output<Node>& input,
+                                       const std::vector<int64_t>& axes) {
+    TENSORFLOW_OP_VALIDATION(
+        node,
+        axes.size() == 0 || axes.size() == 1,
+        "OpenVINO TensorFlow Frontend does not support Reverse and ReverseV2 with multiple axes for the reversing.");
+    auto reverse_v2_node_name = node.get_name();
+
+    if (axes.size() == 0) {
+        // there is nothing to reverse
+        input.get_tensor().add_names({reverse_v2_node_name + ":0"});
+        return {input};
+    }
+
+    int64_t seq_axis = axes[0];
+    int64_t batch_axis = 0;
+
+    // normalize batch_axis and seq_axis so that they become non-negative and different
+    std::vector<int64_t> unsqueeze_axes;
+    if (seq_axis == 0) {
+        // unsqueeze is needed by axis = 0, 1
+        unsqueeze_axes.push_back(0);
+        unsqueeze_axes.push_back(1);
+        seq_axis += 2;
+    } else if (seq_axis < 0) {
+        // for seq_axis is negative, we can handle it only if input rank is defined
+        // since we need it for seq_axis normalization
+        auto input_rank = input.get_partial_shape().rank();
+        if (input_rank.is_static()) {
+            auto input_rank_value = static_cast<int64_t>(input_rank.get_length());
```

Comment:
Why cast? `get_length()` returns `int64_t` by default.

Reply:
Great catch. I thought it to be `size_t`.